decidim/decidim
# Fix decidim-templates usage

#### :tophat: What? Why?
This PR attempts to fix  #12332. 

While investigating the issue i have noticed that `decidim-templates` is a hard dependency at the moment. It is a dependency of decidim-survey. The issue #12332 is caused by the fact that we have a `require "decdim/templates"` that has been introduced in #9607, which has not been backported to 0.26. 

Since #6247 we list the decidim-templates as a hard dependency of decidim-surveys. 
Since #9577 we changed the way we check for a module being installed when using `Decidim.module_installed?` by actually doing a `require "decidim/#{mod}"`, where mod in our case is `templates`, which will be always true, as the result of dependency added in #6247. 
Since #9607 we are automatically require the templates engine as a dependency of surveys. 
Moreover, the fix added in #11613 is calling `Decidim.module_enabled?` which is automatically requires and loads the templates, which causes the menu to be registered. 

Considering the above information, i have added some checks that should be easily backported to 0.26. 

Since this PR is related to gem dependencies that cannot be disabled on spec runtime, i am not 100% what kind of specs i need to add. 

#### Testing
1. Create a development app  and edit the Gemfile 
2. Make sure the decidim-templates is commented out (and run bundle specific)
3. Start the application see the templates menu is still there 
4. Edit `decidim-surveys/lib/decidim/surveys/engine.rb` and comment out the `require "decidim/templates"`
5. restart the application, and visit the admin 
6. In Participatory Process you should get a missing i18n key 
7. You should also have a `Decidim::Templates::Admin::ApplicationHelper` related error 
8. In meetrings / registration / registration form you should see the error reported  in #12332
9. Apply the patch, see that errors from 6, 7, 8 are not replicable. 

:hearts: Thank you!
